pkg/conversions/virtualrouter_types_conversion.go

```diff
@@ -0,0 +1,327 @@
+package conversions
+
+import (
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-sdk-go/aws"
+	appmeshsdk "github.com/aws/aws-sdk-go/service/appmesh"
+	"k8s.io/apimachinery/pkg/conversion"
+)
+
+func Convert_CRD_VirtualRouterListener_To_SDK_VirtualRouterListener(crdObj *appmesh.VirtualRouterListener,
+	sdkObj *appmeshsdk.VirtualRouterListener, scope conversion.Scope) error {
+
+	sdkObj.PortMapping = &appmeshsdk.PortMapping{}
+	return Convert_CRD_PortMapping_To_SDK_PortMapping(&crdObj.PortMapping, sdkObj.PortMapping, scope)
+}
+
+func Convert_CRD_WeightedTarget_To_SDK_WeightedTarget(crdObj *appmesh.WeightedTarget,
+	sdkObj *appmeshsdk.WeightedTarget, scope conversion.Scope) error {
+
+	sdkObj.VirtualNode = aws.String("")
+	if err := scope.Convert(&crdObj.VirtualNodeRef, sdkObj.VirtualNode, scope.Flags()); err != nil {
+		return err
+	}
+	sdkObj.Weight = aws.Int64(crdObj.Weight)
+	return nil
+}
+
+func Convert_CRD_MatchRange_To_SDK_MatchRange(crdObj *appmesh.MatchRange,
+	sdkObj *appmeshsdk.MatchRange, scope conversion.Scope) error {
+	sdkObj.Start = aws.Int64(crdObj.Start)
+	sdkObj.End = aws.Int64(crdObj.End)
+	return nil
+}
+
+func Convert_CRD_HeaderMatchMethod_To_SDK_HeaderMatchMethod(crdObj *appmesh.HeaderMatchMethod,
+	sdkObj *appmeshsdk.HeaderMatchMethod, scope conversion.Scope) error {
+
+	sdkObj.Exact = crdObj.Exact
+	sdkObj.Prefix = crdObj.Prefix
+
+	sdkObj.Range = &appmeshsdk.MatchRange{}
```

Comment:
we need to check crdObj == nil here.
If crdObj == nil, sdkObj.Range should be nil. also(these Convert*'s input should be non-nil)

Reply:
Ah I totally missed that. Thanks for pointing it out. I'll address here and all the other places

Reply:
Done